### PR TITLE
sphinx: install requirements.txt

### PR DESCRIPTION
**Pull request by nickdesaulniers**
Link: https://github.com/llvm/llvm-project/pull/120274
Link: https://github.com/llvm/llvm-project/pull/120319

### CODE REVIEW

File: zorg/buildbot/builders/SphinxDocsBuilder.py [hunk 1]
```diff
@@ -176,6 +176,12 @@ def getSphinxRuntimesDocsBuildFactory(
             src_to_build_dir='runtimes',
             **kwargs) # Pass through all the extra arguments.
 
+    f.addStep(ShellCommand(
+                name="Install pip depenencies",
+                command=['pip install -r llvm-project/llvm/docs/requirements.txt'],
```
Comment (nickdesaulniers) [inline zorg/buildbot/builders/SphinxDocsBuilder.py:181]: This might have to be relative...I probably want `src_dir` instead of `obj_dir`.

Comment (nickdesaulniers) [reply zorg/buildbot/builders/SphinxDocsBuilder.py]: 8184e7fd11a5

File: zorg/buildbot/builders/SphinxDocsBuilder.py [hunk 2]
```diff
@@ -176,6 +176,12 @@ def getSphinxRuntimesDocsBuildFactory(
             src_to_build_dir='runtimes',
             **kwargs) # Pass through all the extra arguments.
 
+    f.addStep(ShellCommand(
+                name="Install pip dependencies",
+                command=['pip install -r llvm/docs/requirements.txt'],
```
Comment (boomanaiden154) [inline zorg/buildbot/builders/SphinxDocsBuilder.py:181]: This should be `llvm/docs/requirements-hashed.txt`.

Comment (nickdesaulniers) [reply zorg/buildbot/builders/SphinxDocsBuilder.py]: 0cc13d4ec1c417c5e22312a50975435cb068f13e

**Review by boomanaiden154 — APPROVED** [review]
LGTM.

Although @gkistanova will need to confirm whether or not this will run properly on the bot.

**Review by marcauberer — APPROVED** [review]

**Comment by nickdesaulniers** [comment]
I'm planning to be on vacation for 2 weeks (plus tomorrow). So merging this now, as I only have today to work through issues if it's not perfect.

**Comment by nickdesaulniers** [comment]
The bots build queue looks stuck to me: https://lab.llvm.org/buildbot/#/builders/178 :shrug:

**Comment by nickdesaulniers** [comment]
Builds are back to green.

**Pull request merged**